cmd/integration_test.go

```diff
@@ -63,31 +54,17 @@ type TestCase struct {
 	ExpectAllow   bool
 	Action        acl.EnforcementPolicy
 	ExpectStatus  int
-	OverTls       bool
+	OverTLS       bool
 	OverConnect   bool
 	ProxyURL      string
-	TargetPort    int
 	RandomTrace   int
-	TargetHost    string
-	TargetScheme  string
+	TargetURL     string
 	RoleName      string
 	UpstreamProxy string
 }
 
-func conformIllegalProxyResult(t *testing.T, test *TestCase, resp *http.Response, err error, logs []*logrus.Entry) {
-	r := require.New(t)
-	a := assert.New(t)
-	t.Logf("HTTP Response: %#v", resp)
-
-	// TODO: fix this after smokescreen's returned errors are improved
-	entry := findLogEntry(logs, "CANONICAL-PROXY-DECISION")
-	r.NotNil(entry)
-	// Make sure the hostname resolution of the upstream failed. This indicates
-	// that the request wasn't forwarded directly to the proxy target.
-	a.Contains(entry.Data["error"], "notaproxy.service")
-}
-
-func conformResult(t *testing.T, test *TestCase, resp *http.Response, err error, logs []*logrus.Entry) {
+// isValidProxyResponse validates tests cases and expected responses from TestSmokescreenIntegration
+func isValidProxyResponse(t *testing.T, test *TestCase, resp *http.Response, err error, logs []*logrus.Entry) {
```

Comment:
Ah, I thought you wanted to use this function in a condition, like `if isValidProxyResponse { ... }`, but if not, I retract my suggestion to call it `is*` (which looks like it should return a boolean).

Reply:
Agreed! I changed these to `validateProxyResponse...`